File: Backend-API/models/rule.py
from datetime import datetime
import uuid

class Rule:
    @staticmethod
    def create(achievement_id, event_type, condition_type, threshold, field=None, category=None):
        rule = {
            '_id': str(uuid.uuid4()),
            'achievement_id': achievement_id,
            'event_type': event_type,
            'condition_type': condition_type,
            'threshold': threshold,
            'field': field,
            'created_at': datetime.now()
        }
        
        # Add category only if provided
        if category is not None:
            rule['category'] = category
            
        return rule

File: RewardXLibrary/src/main/java/com/example/rewardxlibrary/models/Rule.kt
package com.example.rewardxlibrary.models

import com.google.gson.annotations.SerializedName

data class Rule(
    @SerializedName("_id")
    val id: String? = null,

    @SerializedName("achievement_id")
    val achievementId: String,

    @SerializedName("event_type")
    val eventType: String,

    @SerializedName("condition_type")
    val conditionType: String,

    @SerializedName("threshold")
    val threshold: Int,

    @SerializedName("field")
    val field: String? = null,

    @SerializedName("created_at")
    val createdAt: String? = null
)